src/testrel.jl

```diff
@@ -588,7 +588,11 @@ function _test_rel_step(
             end
```

Comment:
I think you're right, unlike the logic.

Reply:
With a new dawn shedding new light I can say that it does give the correct result.

We first check that the expected problems are present. We then search for unexpected errors/exceptions and report them. If we were expecting any problems then we also accept other errors.

Reply:
I don't think I'm following why if we were expecting any problems that unexpected problems are fine. I see that they're reported, but they don't seem to fail the test. I'd think that they should fail the test.

Reply:
The motivation is that you can check for the particular problem of interest, without also having to specify the errors that stem from that problem.

It would be nice to be able to specify that unexpected problems are not allowed, but that wasn't part of the original design. Maybe we could add `expected_problems_only::Bool`?